### PR TITLE
Org picker

### PR DESCRIPTION
**Pull request by RevelsGit**

### CODE REVIEW

File: app/imports/startup/client/routes.js [hunk 1]
```diff
@@ -28,14 +31,32 @@ AccountsTemplates.configureRoute('signUp', {
   path: '/sign-up',
   layoutTemplate: 'LoginLayout',
   layoutRegions: {},
-  contentRegion: 'content'
+  contentRegion: 'content',
+  redirect: redirectHandler
 });
 
-FlowRouter.route('/:orgSerialNumber', {
```
Comment (meyginix) [inline app/imports/startup/client/routes.js]: Url for organization's dashboard should be "/:serialNumber" ("/15645", "/432534", etc), without "organizations".
And also, this this route must be defined last.

File: app/imports/startup/client/routes.js [hunk 2]
```diff
@@ -28,14 +31,32 @@ AccountsTemplates.configureRoute('signUp', {
...
 });
+
+FlowRouter.route('/organizations', {
+  name: 'organizationPickerPage',
+  action(params) {
+    BlazeLayout.render('OrganizationPickerPage');
+  }
+});
+
+function redirectHandler() {
```
Comment (meyginix) [inline app/imports/startup/client/routes.js]: This function runs before subscription for user's organizations is ready.

File: app/imports/startup/client/routes.js [hunk 3]
```diff
@@ -28,14 +31,32 @@ AccountsTemplates.configureRoute('signUp', {
...
+  const organizationsCount = Organizations.find({'users.userId': Meteor.userId()}).count();
+  if (organizationsCount > 1) {
+    FlowRouter.go('organizationPickerPage')
+  } else {
+    const { serialNumber } = Organizations.findOne();
```
Comment (meyginix) [inline app/imports/startup/client/routes.js]: If user is not logged in, there are no organizations in the client-side collection and this line causes error.

File: app/imports/startup/client/routes.js [hunk 4]
```diff
@@ -28,14 +31,32 @@ AccountsTemplates.configureRoute('signUp', {
...
+    FlowRouter.go('dashboardPage', { orgSerialNumber: serialNumber });
+  }
+};
```
Comment (meyginix) [inline app/imports/startup/client/routes.js]: I think we should redirect user to url "/" from "/sign-in" and "/sign-up" if he is logged in. And then on "/" redirect to organization's dashboard or show organization picker. 

@Praxie, @trsdln, @RevelsGit , what do you think?

File: app/imports/startup/client/routes.js [hunk 5]
```diff
@@ -28,14 +31,32 @@ AccountsTemplates.configureRoute('signUp', {
...
   name: 'dashboardPage',
   action(params) {
     BlazeLayout.render('DashboardLayout', {
       content: 'DashboardPage'
     });
   }
 });
+
+FlowRouter.route('/organizations', {
+  name: 'organizationPickerPage',
```
Comment (meyginix) [inline app/imports/startup/client/routes.js]: I think this page should use DashboardLayout.

@Praxie, @trsdln, @RevelsGit what do you think?

File: app/imports/startup/client/routes.js (hunk 4 above)
Comment (Praxie) [inline app/imports/startup/client/routes.js]: @meyginix @RevelsGit @trsdln As we discussed today, we'll need to do this the same way Basecamp and Evernote do. So the landing page "/" should be the same all the time. It should contain both Sign In and Sign Up buttons for both signed in and not signed in users. For signed in users the Sign In button should redirect to the Dashboard of the organization that the user is the owner of. (User can't be an owner of multiple organizations). If the user doesn't own any organizations, then he should be redirected to the latest selected organization.
@RevelsGit, so we agreed to remove the whole organization picker template.
@meyginix, @trsdln, @RevelsGit What is the best way to make all these redirects? Do we need to create a special route? What do you guys think?

Comment (Praxie) [inline app/imports/startup/client/routes.js]: @meyginix @trsdln, So @RevelsGit proposed to use AccountsTemplates hooks to handle the redirection.

File: app/imports/startup/client/routes.js (hunk 1 above)
Comment (RevelsGit) [inline app/imports/startup/client/routes.js]: Then all route paths like /sign-in, /actions etc will be considered by router as orgSerialNumber

File: app/imports/startup/client/routes.js (hunk 4 above)
Comment (trsdln) [inline app/imports/startup/client/routes.js]: @meyginix @Praxie @RevelsGit I think we should do it as simple as possible:

* create url like `/launch-app` which is going to redirect user to the last organization he was working in.

* if user doesn't have last organization render select organization picker we already have OR  some help text that describes how to select organization.

* set last selected organization as new created organization if user just created one.

@Praxie, Does it makes sense?

File: app/imports/startup/client/routes.js (hunk 1 above)
Comment (meyginix) [inline app/imports/startup/client/routes.js]: That's why it must be defined last.

File: app/imports/startup/client/routes.js (hunk 4 above)
Comment (Praxie) [inline app/imports/startup/client/routes.js]: @trsdln, 
> if user doesn't have last organization render select organization picker we already have OR some help text that describes how to select organization.

I think that findOne() will work just fine in this situation.

**Comment by Praxie** [comment]
The main organization should have the highest priority. If the user owns an organization, then he should be redirected to it every time no matter what the selected org is.

**Pull request closed without merging**